### ISSUE TITLE
Starting bitcoin-qt with -nowallet and then opening a wallet does not show the wallet

### ISSUE THREAD

**Issue by achow101** [bitcoin/bitcoin#15453]
When starting bitcoin-qt with `-nowallet` to explicitly not load any wallets initially, attempting to load a wallet using the Open Wallet menu option results in the wallet being loaded but the GUI does not change. The wallet is reachable via the RPC interface but the GUI still says that no wallets are loaded.

**Comment by promag** [comment]
Should tag with 0.18.

**Comment by promag** [comment]
@achow101 I fail to reproduce, mind sharing more details, Qt version etc.

**Comment by jnewbery** [comment]
I tested this exact scenario in the PR that added the "open wallet" menu and it worked for me.

**Comment by promag** [comment]
Also tested with #15450 and still works.

**Comment by achow101** [comment]
Bitcoin Core version v0.17.99.0-f9775a8655 (64-bit)
Qt version 5.7.1
Using Debian 9

**Comment by achow101** [comment]
I built it using the depends system and this bug is not present. It seems that the issue has to do with Qt 5.7.1.

**Comment by fanquake** [comment]
Can confirm this issue. I compiled master inside a new Debian 9.8 vm (qt 5.7.1), a new wallet opened with the Open wallet menu doesn't display in the GUI.

![debian 9 8](https://user-images.githubusercontent.com/863730/53143654-ec47a280-35d3-11e9-8774-e814218cbbda.gif)

**Comment by promag** [comment]
@fanquake mind testing with same fix of #15455?

**Comment by achow101** [comment]
I've tried with that patch but it does not fix this.

**Comment by jonasschnelli** [comment]
I was not able to reproduce this, not on OSX self compiled master, not on Ubuntu 18.04.1 with a nightly build (gitian build of master) nor with self compiled master on Ubuntu (Qt 5.9.5). Either a timing issue or a Qt upstream difference.

**Comment by Sjors** [comment]
My guess is that this is a QT 5.7 issue. @achow101 does it go away when you use the depends 5.9 version?

**Comment by promag** [comment]
@Sjors yes, in IRC he said it works.

**Comment by promag** [comment]
I'm working on a fix.

**Comment by fanquake** [comment]
@promag mentioned [(line 197)](http://www.erisian.com.au/bitcoin-core-dev/log-2019-03-01.html) that #15478 is a potential fix for this issue.

**Comment by fanquake** [comment]
Moving this to the [`0.18.1` milestone](https://github.com/bitcoin/bitcoin/milestone/41).

**Comment by promag** [comment]
@fanquake care to test with #15478?

**Comment by fanquake** [comment]
@promag is #16261 a potential fix for this?

**Comment by promag** [comment]
@fanquake yes I think so.

**Comment by hebasto** [comment]
Can confirm this bug on Debian 9.9 (Qt 5.7.1, master 8c69fae94410f54bad13be0f34d54370fddbf4b3).

Debugging info:
```
hebasto@debian:~/bitcoin$ QT_FATAL_WARNINGS=1 lldb ./src/qt/bitcoin-qt -- -regtest -nowallet -debug=qt
(lldb) target create "./src/qt/bitcoin-qt"
Current executable set to './src/qt/bitcoin-qt' (x86_64).
(lldb) settings set -- target.run-args  "-regtest" "-nowallet" "-debug=qt"
(lldb) run
Process 890 launched: './src/qt/bitcoin-qt' (x86_64)
Process 890 stopped
* thread #17: tid = 912, 0x00007ffff400bfff libc.so.6`gsignal + 207, name = 'QThread', stop reason = signal SIGABRT
    frame #0: 0x00007ffff400bfff libc.so.6`gsignal + 207
libc.so.6`gsignal:
->  0x7ffff400bfff <+207>: addq   $0x88, %rsp
    0x7ffff400c006 <+214>: movl   %r8d, %eax
    0x7ffff400c009 <+217>: popq   %rbx
    0x7ffff400c00a <+218>: retq   
(lldb) bt
error: bitcoin-qt Parsing a die that is being parsed die: 0x0112da0e: DW_TAG_subprogram getOrCreateWallet
error: bitcoin-qt 0x010f864e: DW_TAG_member '_M_local_buf' refers to type 0x011168a2 which extends beyond the bounds of 0x010f8646
* thread #17: tid = 912, 0x00007ffff400bfff libc.so.6`gsignal + 207, name = 'QThread', stop reason = signal SIGABRT
  * frame #0: 0x00007ffff400bfff libc.so.6`gsignal + 207
    frame #1: 0x00007ffff400d42a libc.so.6`abort + 362
    frame #2: 0x00007ffff632b895 libQt5Core.so.5`QMessageLogger::warning(char const*, ...) const + 341
    frame #3: 0x00007ffff6520179 libQt5Core.so.5`QMetaMethod::invoke(QObject*, Qt::ConnectionType, QGenericReturnArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument) const + 2569
    frame #4: 0x00007ffff65258aa libQt5Core.so.5`QMetaObject::invokeMethod(QObject*, char const*, Qt::ConnectionType, QGenericReturnArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument) + 2410
    frame #5: 0x000055555565abb8 bitcoin-qt`WalletController::getOrCreateWallet(std::unique_ptr<interfaces::Wallet, std::default_delete<interfaces::Wallet> >) [inlined] QMetaObject::invokeMethod(val9=<unavailable>, val8=<unavailable>, val7=<unavailable>, val6=<unavailable>, val5=<unavailable>, val4=<unavailable>, val3=<unavailable>, val2=<unavailable>, val1=<unavailable>, val0=<unavailable>, type=QueuedConnection, member="", obj=0x00005555571f1aa0) + 63 at qobjectdefs.h:461
    frame #6: 0x000055555565ab79 bitcoin-qt`WalletController::getOrCreateWallet(this=0x00005555571f1aa0, wallet=unique_ptr<interfaces::Wallet, std::default_delete<interfaces::Wallet> > @ 0x00007fffb1ff8ae0) + 809 at walletcontroller.cpp:124
    frame #7: 0x000055555565ad2e bitcoin-qt`std::_Function_handler<void(std::unique_ptr<interfaces::Wallet, std::default_delete<interfaces::Wallet> >), WalletController::WalletController(interfaces::Node&, const PlatformStyle*, OptionsModel*, QObject*)::<lambda(std::unique_ptr<interfaces::Wallet, std::default_delete<interfaces::Wallet> >)> >::_M_invoke(const std::_Any_data &const, std::unique_ptr<interfaces::Wallet, std::default_delete<interfaces::Wallet> > &&const) + 12 at walletcontroller.cpp:25
```

debug.log:
```
...
2019-07-06T12:34:51Z GUI: QMetaMethod::invoke: Unable to handle unregistered datatype 'WalletModel*'
```

It seems the problem is here:
https://github.com/bitcoin/bitcoin/blob/8c69fae94410f54bad13be0f34d54370fddbf4b3/src/qt/walletcontroller.cpp#L118-L125

@fanquake I failed to build the Bitcoin Core with the [downloaded Qt 5.7.1](https://download.qt.io/archive/qt/5.7/5.7.1/) on bionic. Could you give me a piece of advice?

**Comment by promag** [comment]
Fixed by #16348?

**Comment by hebasto** [comment]
@promag 
> Fixed by #16348?

Yes.
